src/test/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacadeIT.java

```diff
@@ -432,4 +430,62 @@ public void assignBulkEndpoint_scheduleSingleAssignmentWithScheduledDateAfterDue
         // check status code is NO_CONTENT (successful)
         assertEquals(Response.Status.NO_CONTENT.getStatusCode(), deleteAssignmentResponse.getStatus());
     }
+
+    @Test public void getAssignmentProgressDownloadCSV_succeeds() throws Exception {
+        // log in as Test teacher, create request
+        LoginResult teacherLogin = loginAs(httpSession, ITConstants.TEST_TEACHER_EMAIL,
+                ITConstants.TEST_TEACHER_PASSWORD);
+        HttpServletRequest downloadAssignmentRequest = createRequestWithCookies(new Cookie[]{teacherLogin.cookie});
+        replay(downloadAssignmentRequest);
+
+        Response downloadAssignmentResponse = assignmentFacade.getAssignmentProgressDownloadCSV(downloadAssignmentRequest, 4L, "excel");
+        String downloadAssignmentContents = downloadAssignmentResponse.getEntity().toString();
+
+        try (FileInputStream expectedFile = new FileInputStream("src/test/resources/expected_assignment_progress_export.csv")) {
+            String expectedContents = IOUtils.toString(expectedFile, StandardCharsets.UTF_8);
+
+            assertEquals(expectedContents, downloadAssignmentContents);
+        }
+    }
+
+    @Test public void getAssignmentProgressDownloadCSV_permissionDenied() throws Exception {
+        // log in as Test teacher, create request
+        LoginResult teacherLogin = loginAs(httpSession, ITConstants.TEST_TUTOR_EMAIL,
+                ITConstants.TEST_TUTOR_PASSWORD);
+        HttpServletRequest downloadAssignmentRequest = createRequestWithCookies(new Cookie[]{teacherLogin.cookie});
+        replay(downloadAssignmentRequest);
+
+        Response downloadAssignmentResponse = assignmentFacade.getAssignmentProgressDownloadCSV(downloadAssignmentRequest, 4L, "excel");
+
+        assertEquals(403, downloadAssignmentResponse.getStatus());
+    }
+
+    @Test public void getGroupAssignmentsProgressDownloadCSV_succeeds() throws Exception {
```

Comment:
Now that you've added some tests for the group assignments and they look pretty much the same as the regular assignment, this could be a good way to get some experience with data providers, if you've never used them before. They would get rid of a lot of the repetition. But if it's too much hassle, you can also leave it as it is.